.github/workflows/build-all-on-pr.yml

```diff
@@ -0,0 +1,51 @@
+###########################################################################
+# GitHub Action Workflow
+# On pull request or changes to scripts/commands to any branch, 
+# triggers the full build and test pipeline.
+#
+# Step 0: make PR from your branch into develop, or make changes on your 
+#         branch to non-documentation files in scripts or commands.sh and 
+#         push changes to your remote branch.
+#
+# Step 1: Makes a request to the V2 CircleCI API to initiate the project,
+#         which will filter based upon build_and_test_backend and 
+#         build_and_test_frontend to run the workflow/jobs listed here:
+#           build-and-test:[
+#             test-backend,
+#             test-frontend,
+#             test-e2e
+#           ]
+#
+#         Leverages the open source GitHub Action:
+#           https://github.com/promiseofcake/circleci-trigger-action
+###########################################################################
+name: Build and test All for PRs and when scripts/commands change
+on:
+  push:
+    branches_ignore:
+    - develop
```

Comment:
Is this just for testing? I noted it on all the github actions. This would be removed ultimately or does develop follow some other process I can't see from these diffs?

Reply:
So this build all has two components. on push to any branch except develop it will build all if changes to scripts/commands happen but shouldn't need to happen for develop because of the next part. 

On any PR besides documentation PRs (paths_ignore) it will run the full set of tests. If that PR goes through, we don't want to run the exact same tests once the merge goes through, which counts as a push, I believe.

Reply:
I have the first part set up to build all on changes to scripts/* and commands.sh because I believe other parts of the pipelines and maybe even container builds might reference these? I'm not sure on this yet and haven't looked deeper.

Reply:
changes to the develop branch should trigger the Deploy Develop on PR Merge action which should kick off a deploy and then an end to end integration test on the develop environment. The idea being is that the individual parts are tested during the PR with my above changes, and the integration tests are run against the whole environment after merge and deploy. 

Down the line, it would be nice if the post-deploy integration tests fail, we run a rollback as part of the pipeline.